litestar/_openapi/schema_generation/schema.py

```diff
@@ -358,17 +357,37 @@ def create_schema_for_annotation(annotation: Any) -> Schema:
     return Schema()
 
 
-@dataclass(frozen=True)
 class SchemaCreator:
     __slots__ = ("generate_examples", "plugins", "schemas", "prefer_alias")
-    generate_examples: bool
-    """Whether to generate examples if none are given."""
-    plugins: list[OpenAPISchemaPluginProtocol]
-    """A list of plugins."""
-    schemas: dict[str, Schema]
-    """A mapping of namespaces to schemas - this mapping is used in the OA components section."""
-    prefer_alias: bool
-    """Whether to prefer the alias name for the schema."""
+
+    def __init__(
+        self,
+        generate_examples: bool = False,
+        plugins: list[OpenAPISchemaPluginProtocol] | None = None,
+        schemas: dict[str, Schema] | None = None,
+        prefer_alias: bool = True,
+    ):
+        """Instantiate a SchemaCreator.
+
+        Args:
+            generate_examples: Whether to generate examples if none are given.
+            plugins: A list of plugins.
+            schemas: A mapping of namespaces to schemas - this mapping is used in the OA components section.
+            prefer_alias: Whether to prefer the alias name for the schema.
+        """
+        self.generate_examples = generate_examples
+        self.plugins = plugins if plugins is not None else []
+        self.schemas = schemas if schemas is not None else {}
+        self.prefer_alias = prefer_alias
+
+    @property
+    def not_generating_examples(self) -> SchemaCreator:
```

Comment:
this will be wasteful memory wise - we dont need an instance for this. see my other comment.